src/main/webapp/script.js

```diff
@@ -48,6 +62,9 @@ document.addEventListener('DOMContentLoaded', async () => {
  */
 async function handleImageUpload(event) {
   await validateImageUpload().then(async () => {
+    // Refresh the cache.
+    preloadPhotos();
```

Comment:
At this point we didn't send the new image to the server - it is sent in getBlurAreas function.

Reply:
Right, moved it to the bottom of the function.